uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -219,6 +226,35 @@ define([
       };
 
       /**
+       * Checks if user can see file list in files tab. It needs to check for
+       * soft delete flag on leave request as well for files.
+       *
+       * @return {Boolean} true is user can see file list
+       */
+      this.canShowFileList = function () {
+        var filesWithOutSoftDelete = _.filter(this.request.files, function (file) {
+          return !file.toBeDeleted;
+        });
+
+        return !!this.request.fileUploader.queue.length || !!filesWithOutSoftDelete.length;
```

Comment:
`length` is always 0, or some other value. So !! is not required here for both conditions.